netx/conn.go

```diff
@@ -0,0 +1,138 @@
+package netx
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/sirupsen/logrus"
+)
+
+// InstanceProvider is the channel where updated list of desired service are published
+type InstanceProvider <-chan []Address
+
+// Sender sends payload to given Address
+type Sender func(addr Address, payload []byte) (int, error)
```

Comment:
Maybe change the name to `SenderFunc` so it will be visible that this is a function type.